corehq/apps/app_manager/xpath.py

```diff
@@ -405,9 +405,13 @@ def id(self):
 
 
 class SearchSelectedCasesInstanceXpath(InstanceXpath):
-    id = "search_selected_cases"
+    default_id = "search_selected_cases"
     path = "results/value"
 
+    @property
+    def id(self):
+        return self or self.default_id
```

Comment:
I'm confused by this returning either an XPath or a string. When would self ever be falsy?

Reply:
These classes inherit from string `class XPath(str):` so if you construct one with no arguments then `self` will be like an empty string:

```python
>> SearchSelectedCasesInstanceXpath()
''
```

With this change the `id` field can be adjusted by passing in a different value to the constructor:

```python
>> SearchSelectedCasesInstanceXpath().id
'search_selected_cases'
>> SearchSelectedCasesInstanceXpath("bob").id
'bob'
```

This follows the pattern of `IndicatorXpath`: https://github.com/dimagi/commcare-hq/blob/63bacada96c9fd9e6309e5687bf297d3fdac14b5/corehq/apps/app_manager/xpath.py#L399-L404

Reply:
aha, I had forgotten XPath objects are ultimately strings. Thank you for explaining, that all makes sense.